### PR TITLE
Add aten::silu and their variants 

### PR DESCRIPTION
**Pull request by yucai-intel**

### CODE REVIEW

File: src/aten/sycl/ActivationSiluKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace at {
+namespace native {
+namespace xpu {
+
+template <typename scalar_t>
+struct SiluFunctor {
+  scalar_t operator()(scalar_t x) const {
+    using opmath_t = at::opmath_type<scalar_t>;
+          const opmath_t x_acc = static_cast<opmath_t>(x);
+          return x_acc / (opmath_t(1) + ::exp(-x_acc));
```
Comment (fengyuan14) [inline src/aten/sycl/ActivationSiluKernel.cpp:18]: Let us using std::exp.

File: src/aten/sycl/ActivationSiluKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+template <typename scalar_t>
+struct SiluBackwardFunctor {
+  scalar_t operator()(scalar_t dy, scalar_t x) const {
+    using opmath_t = at::opmath_type<scalar_t>;
+          const opmath_t dy_acc = static_cast<opmath_t>(dy);
+          const opmath_t x_acc = static_cast<opmath_t>(x);
+          const opmath_t s_acc =
+              opmath_t(1) / (opmath_t(1) + c10::cuda::compat::exp(-x_acc));
```
Comment (fengyuan14) [inline src/aten/sycl/ActivationSiluKernel.cpp:29]: std::exp

**Comment by fengyuan14** [comment]
Please add fine gran cases.

File: src/aten/sycl/ActivationSiluKernel.cpp (hunk 2 above)
Comment (yucai-intel) [reply src/aten/sycl/ActivationSiluKernel.cpp]: Done.

**Comment by fengyuan14** [comment]
image

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**